### PR TITLE
Tokenizers: ability to load from model subfolder

### LINKED ISSUE #8447: Model name 'facebook/rag-sequence-base/*' not found when running examples/rag/finetune.sh
## Environment info
     
- `transformers` version: 3.3.1
- Platform: Linux-4.15.0-38-generic-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.6.9
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: True
- Using distributed or parallel set-up in script?: True (Retriever is distributed)

### Who can help
@patrickvonplaten, @lhoestq 

## Information

Model I am using (Bert, XLNet ...):

**facebook/rag-sequence-base**

The problem arises when using:
* [x ] the official example scripts: (give details below)
examples/rag/finetune.sh

The tasks I am working on is:
* [x ] my own task or dataset: (give details below)

## To reproduce

run `sh finetune.sh`
with 
```
DATA_DIR=data_dir
OUTPUT_DIR=output_dir
MODEL_NAME_OR_PATH="facebook/rag-sequence-base"
```

gives:

**Model name 'facebook/rag-sequence-base/question_encoder_tokenizer' not found in model shortcut name list (facebook/dpr-question_encoder-single-nq-base). Assuming 'facebook/rag-sequence-base/question_encoder_tokenizer' is a path, a model identifier, or url to a directory containing tokenizer files**.
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/question_encoder_tokenizer/vocab.txt from cache at /h/asabet/.cache/torch/transformers/14d599f015518cd5b95b5d567b8c06b265dbbf04047e44b3654efd7cbbacb697.542ce4285a40d23a559526243235df47c5f75c197f04f37d1a0c124c32c9a084
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/question_encoder_tokenizer/added_tokens.json from cache at None
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/question_encoder_tokenizer/special_tokens_map.json from cache at /h/asabet/.cache/torch/transformers/70614c7a84151409876eaaaecb3b5185213aa5c560926855e35753b9909f1116.275045728fbf41c11d3dae08b8742c054377e18d92cc7b72b6351152a99b64e4
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/question_encoder_tokenizer/tokenizer_config.json from cache at /h/asabet/.cache/torch/transformers/8ade9cf561f8c0a47d1c3785e850c57414d776b3795e21bd01e58483399d2de4.11f57497ee659e26f830788489816dbcb678d91ae48c06c50c9dc0e4438ec05b
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/question_encoder_tokenizer/tokenizer.json from cache at None
**Model name 'facebook/rag-sequence-base/generator_tokenizer' not found in model shortcut name list (facebook/bart-base, facebook/bart-large, facebook/bart-large-mnli, facebook/bart-large-cnn, facebook/bart-large-xsum, yjernite/bart_eli5). Assuming 'facebook/rag-sequence-base/generator_tokenizer' is a path, a model identifier, or url to a directory containing tokenizer files.**
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/generator_tokenizer/vocab.json from cache at /h/asabet/.cache/torch/transformers/3b9637b6eab4a48cf2bc596e5992aebb74de6e32c9ee660a27366a63a8020557.6a4061e8fc00057d21d80413635a86fdcf55b6e7594ad9e25257d2f99a02f4be
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/generator_tokenizer/merges.txt from cache at /h/asabet/.cache/torch/transformers/b2a6adcb3b8a4c39e056d80a133951b99a56010158602cf85dee775936690c6a.70bec105b4158ed9a1747fea67a43f5dee97855c64d62b6ec3742f4cfdb5feda
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/generator_tokenizer/added_tokens.json from cache at None
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/generator_tokenizer/special_tokens_map.json from cache at /h/asabet/.cache/torch/transformers/342599872fb2f45f954699d3c67790c33b574cc552a4b433fedddc97e6a3c58e.6e217123a3ada61145de1f20b1443a1ec9aac93492a4bd1ce6a695935f0fd97a
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/generator_tokenizer/tokenizer_config.json from cache at /h/asabet/.cache/torch/transformers/e5f72dc4c0b1ba585d7afb7fa5e3e52ff0e1f101e49572e2caaf38fab070d4d6.d596a549211eb890d3bb341f3a03307b199bc2d5ed81b3451618cbcb04d1f1bc
loading file https://s3.amazonaws.com/models.huggingface.co/bert/facebook/rag-sequence-base/generator_tokenizer/tokenizer.json from cache at None
Traceback (most recent call last):
  File "finetune.py", line 499, in <module>
    main(args)
  File "finetune.py", line 439, in main
    model: GenerativeQAModule = GenerativeQAModule(args)
  File "finetune.py", line 105, in __init__
    retriever = RagPyTorchDistributedRetriever.from_pretrained(hparams.model_name_or_path, config=config)
  File "/h/asabet/.local/lib/python3.6/site-packages/transformers/retrieval_rag.py", line 308, in from_pretrained
    config, question_encoder_tokenizer=question_encoder_tokenizer, generator_tokenizer=generator_tokenizer
  File "/scratch/ssd001/home/asabet/transformers/examples/rag/distributed_retriever.py", line 41, in __init__
    index=index,
**TypeError: __init__() got an unexpected keyword argument 'index'**

## Expected behavior
finetune.sh should launch and run

### PR DESCRIPTION
**Pull request by julien-c**
Should fix #8447

### CODE REVIEW

File: src/transformers/models/auto/tokenization_auto.py [hunk 1]
```diff
@@ -250,10 +250,9 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
             pretrained_model_name_or_path (:obj:`str`):
                 Can be either:
 
-                    - A string with the `shortcut name` of a predefined tokenizer to load from cache or download, e.g.,
-                      ``bert-base-uncased``.
-                    - A string with the `identifier name` of a predefined tokenizer that was user-uploaded to our S3,
-                      e.g., ``dbmdz/bert-base-german-cased``.
+                    - A string, the `model id` of a predefined tokenizer hosted inside a model repo on huggingface.co.
+                      Valid model ids can be located at the root-level, like ``bert-base-uncased``, or namespaced under
+                      a user or organization name, like ``dbmdz/bert-base-german-cased``.
```
Comment (julien-c) [inline src/transformers/models/auto/tokenization_auto.py:255]: I apologize for this change that's orthogonal to this PR. Will propagate this change everywhere if the wording is agreed upon.

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -872,6 +874,9 @@ def hf_bucket_url(model_id: str, filename: str, revision: Optional[str] = None,
     its sha1 if stored in git, or its sha256 if stored in git-lfs. Files cached locally from transformers before v3.5.0
     are not shared with those new files, because the cached file's name contains a hash of the url (which changed).
     """
+    if subfolder is not None:
+        filename = f"{subfolder}/{filename}"
+
```
Comment (julien-c) [inline src/transformers/file_utils.py:879]: looks like Python strings are value types so this should be safe @sgugger

Comment (sgugger) [reply src/transformers/file_utils.py:879]: Yes, I have learned my lesson :-)

File: src/transformers/models/auto/tokenization_auto.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/models/auto/tokenization_auto.py:255]: This makes more sense so it's good to go for me!

**Review by sgugger — APPROVED** [review]
LGTM, thanks for adding this functionality!

File: src/transformers/file_utils.py (hunk 2 above)
Comment (julien-c) [reply src/transformers/file_utils.py:879]: (i wasn't sure either – always a good reflex to check I guess)

**Review by lhoestq — APPROVED** [review]
LGTM :)
thanks for fixing it

File: examples/rag/finetune.sh [hunk 3]
```diff
@@ -7,7 +7,7 @@ export PYTHONPATH="../":"${PYTHONPATH}"
 python examples/rag/finetune.py \
     --data_dir $DATA_DIR \
     --output_dir $OUTPUT_DIR \
-    --model_name_or_path $MODLE_NAME_OR_PATH \
+    --model_name_or_path $MODEL_NAME_OR_PATH \
```
Comment (sgugger) [inline examples/rag/finetune.sh:10]: Nice catch!

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Review by patrickvonplaten — APPROVED** [review]
Great!

**Pull request merged**